src/autoscaler/metricscollector/cmd/metricscollector/main.go

```diff
@@ -159,6 +176,65 @@ func main() {
 	logger.Info("exited")
 }
 
+func initDBLockRunner(conf *config.Config, logger lager.Logger, lockDB db.LockDB) ifrit.Runner {
+	dbLockMaintainer := ifrit.RunFunc(func(signals <-chan os.Signal, ready chan<- struct{}) error {
+		ttl := conf.DBLock.LockTTL
+		lockTicker := time.NewTicker(ttl)
+		readyFlag := true
+		owner := getOwner(logger, conf)
+		if owner == "" {
+			logger.Info("failed-to-get-owner-details")
+			os.Exit(1)
+		}
+		lock := &models.Lock{Owner: owner, Ttl: ttl}
+		isLockAcquired, lockErr := lockDB.Lock(lock)
+		if lockErr != nil {
+			logger.Error("failed-to-acquire-lock-in-first-attempt", lockErr)
+		}
+		if isLockAcquired {
+			logger.Info("lock-acquired-in-first-attempt", lager.Data{"owner": owner, "isLockAcquired": isLockAcquired})
+			close(ready)
+			readyFlag = false
+		}
+		for {
+			select {
+			case <-signals:
+				logger.Info("received-interrupt-signal", lager.Data{"owner": owner})
+				lockTicker.Stop()
+				releaseErr := lockDB.Release(owner)
+				if releaseErr != nil {
+					logger.Error("failed-to-release-lock ", releaseErr)
+				} else {
+					logger.Info("successfully-released-lock", lager.Data{"owner": owner})
+				}
+				readyFlag = true
+				return nil
+
+			case <-lockTicker.C:
+				logger.Info("retry-acquiring-lock", lager.Data{"owner": owner})
+				lock := &models.Lock{Owner: owner, Ttl: ttl}
+				isLockAcquired, lockErr := lockDB.Lock(lock)
+				if lockErr != nil {
+					logger.Error("failed-to-acquire-lock", lockErr)
+					releaseErr := lockDB.Release(owner)
+					if releaseErr != nil {
+						logger.Error("failed-to-release-lock ", releaseErr)
+					} else {
+						logger.Info("successfully-released-lock", lager.Data{"owner": owner})
+					}
+					os.Exit(1)
+				}
+				if isLockAcquired && readyFlag {
+					close(ready)
+					readyFlag = false
+					logger.Info("successfully-acquired-lock", lager.Data{"owner": owner})
+				}
+			}
+		}
+	})
+	return dbLockMaintainer
+}
```

Comment:
if only  Lock() and Release() are used,  why we have 	following as Public
 Fetch() (*models.Lock, error)
 Acquire(lock *models.Lock) error
 Renew(owner string) error

Reply:
by the way,  I guess we can define a lock interface, and then use DBLock as one of the implementation.